libsolidity/ast/Types.cpp

```diff
@@ -889,6 +889,9 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 		case Literal::SubDenomination::Wei:
 		case Literal::SubDenomination::Second:
 			break;
+		case Literal::SubDenomination::Gwei:
+			value *= bigint(100'0000'000);
```

Comment:
Please either correctly or not at all. Also check that it works on all platforms and use something like `"100""000"` otherwise.